paddle_billing/Resources/Adjustments/Operations/CreateAdjustment.py

```diff
@@ -1,15 +1,58 @@
 from dataclasses import dataclass
 
+from paddle_billing.Undefined import Undefined
+
 from paddle_billing.Operation import Operation
 
-from paddle_billing.Entities.Shared import Action
+from paddle_billing.Entities.Shared import Action, AdjustmentActionType
 
 from paddle_billing.Resources.Adjustments.Operations import CreateAdjustmentItem
 
+from paddle_billing.Exceptions.SdkExceptions.InvalidArgumentException import InvalidArgumentException
+
 
 @dataclass
 class CreateAdjustment(Operation):
     action: Action
-    items: list[CreateAdjustmentItem]
+    items: list[CreateAdjustmentItem] | None | Undefined
```

Comment:
`items` cannot be set to `Undefined()` by default without moving the argument, which would be a breaking change.

`CreateAdjustment.full()` and `CreateAdjustment.partial()` have been introduced to make it easier to create this operation, e.g.

```python
CreateAdjustment.partial(
    Action.Refund,
    [CreateAdjustmentItem("txnitm_01h8bxryv3065dyh6103p3yg28", AdjustmentType.Partial, "100")],
    "error",
    "txn_01h8bxpvx398a7zbawb77y0kp5",
)
```

```python
CreateAdjustment.full(
    Action.Refund,
    "error",
    "txn_01h8bxpvx398a7zbawb77y0kp5",
)
```

The constructor can still be used, but items must always be provided, e.g.
```python
CreateAdjustment(
    Action.Refund,
    None,
    "error",
    "txn_01h8bxpvx398a7zbawb77y0kp5",
    AdjustmentActionType.Full,
)
```

```python
CreateAdjustment(
    Action.Refund,
    Undefined(),
    "error",
    "txn_01h8bxpvx398a7zbawb77y0kp5",
    AdjustmentActionType.Full,
)
```